assembl/models/auth.py

```diff
@@ -1989,3 +2128,34 @@ def __repr__(self):
                 LanguagePreferenceOrder(self.source_of_evidence).name,
                 self.preferred_order or 0
             )
+
+
+class PostUserLanguagePreference(UserLanguagePreference):
```

Comment:
I think we need to set up a centralized system for user preferences. Currently, for the V1 (and the V2), we have models.preferences.Preferences (Json) and UserPreferenceCollection a wrapper for reading and writing data. This system is centralized, extensible and maintainable but is not sufficiently expressive (for complex needs). We also have an ad hoc system. In this case the preferences are decentralized and not easily maintainable but sufficiently expressive. I think we need to have a centralized system for user preference. For example, a  table of preferences 'Preference' (not Json but can contain a Json) to describe complex needs. In this table we add the column 'language_preference' (for your need) which references an entry in a table 'Translation' and another column 'harvesting_translation' which references another entry in the same table 'Translation'. And we can add the Cookies preferences in this 'Preference' table.